ibm/resource_ibm_app_config_feature_test.go

```diff
@@ -0,0 +1,141 @@
+	nameUpdate := fmt.Sprintf("tf_name_%d", acctest.RandIntRange(10, 100))
+	tags := fmt.Sprintf("tags_%d", acctest.RandIntRange(10, 100))
+	tagsUpdated := fmt.Sprintf("tags_updated_%d", acctest.RandIntRange(10, 100))
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckIbmAppConfigFeatureDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccCheckIbmAppConfigFeatureConfigBasic(instanceName, name, featureID, featureType, description, tags),
+				Check: resource.ComposeAggregateTestCheckFunc(
+					testAccCheckIbmAppConfigFeatureExists("ibm_app_config_feature.ibm_app_config_feature_resource1", conf),
+					resource.TestCheckResourceAttrSet("ibm_app_config_feature.ibm_app_config_feature_resource1", "id"),
+					resource.TestCheckResourceAttrSet("ibm_app_config_feature.ibm_app_config_feature_resource1", "type"),
+					resource.TestCheckResourceAttrSet("ibm_app_config_feature.ibm_app_config_feature_resource1", "name"),
+					resource.TestCheckResourceAttrSet("ibm_app_config_feature.ibm_app_config_feature_resource1", "tags"),
+					resource.TestCheckResourceAttrSet("ibm_app_config_feature.ibm_app_config_feature_resource1", "feature_id"),
+					resource.TestCheckResourceAttrSet("ibm_app_config_feature.ibm_app_config_feature_resource1", "description"),
+					resource.TestCheckResourceAttrSet("ibm_app_config_feature.ibm_app_config_feature_resource1", "enabled_value"),
+					resource.TestCheckResourceAttrSet("ibm_app_config_feature.ibm_app_config_feature_resource1", "disabled_value"),
+				),
+			},
+			{
+				Config: testAccCheckIbmAppConfigFeatureConfigBasic(instanceName, nameUpdate, featureID, featureType, descriptionUpdate, tagsUpdated),
+				Check: resource.ComposeAggregateTestCheckFunc(
+					resource.TestCheckResourceAttr("ibm_app_config_feature.ibm_app_config_feature_resource1", "name", nameUpdate),
+					resource.TestCheckResourceAttr("ibm_app_config_feature.ibm_app_config_feature_resource1", "tags", tagsUpdated),
+					resource.TestCheckResourceAttr("ibm_app_config_feature.ibm_app_config_feature_resource1", "description", descriptionUpdate),
+				),
+			},
+		},
+	})
+}
+
+func testAccCheckIbmAppConfigFeatureConfigBasic(name, envName, featureID, featureType, description, tags string) string {
+	return fmt.Sprintf(`
+		resource "ibm_resource_instance" "app_config_terraform_test456" {
+			name     = "%s"
+			location = "us-south"
+			service  = "apprapp"
+			plan     = "standard"
+		}
+		resource "ibm_app_config_feature" "ibm_app_config_feature_resource1" {
+			guid           	= ibm_resource_instance.app_config_terraform_test456.guid
+			name           	= "%s"
+			environment_id  = "dev"
+			feature_id     	= "%s"
+			type           	= "%s"
+			enabled_value  	= true
+			disabled_value 	= false
+			description    	= "%s"
+			tags    			 	= "%s"
+		}`, name, envName, featureID, featureType, description, tags)
```

Comment:
Can you please add more testcases which covers optional fields also

Reply:
I did not add optional fields because that requires two other resources. We'll implement these two resources next week, will update the test case accordingly.